### ISSUE TITLE
Support default values for WebIDL unions

### ISSUE THREAD

**Issue by ferjm** [servo/servo#21342]
Generating bindings for this dictionary https://webaudio.github.io/web-audio-api/#dictdef-audiocontextoptions breaks with:

```
--- stderr
Traceback (most recent call last):
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/pythonpath.py", line 61, in <module>
    main(sys.argv[1:])
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/pythonpath.py", line 52, in main
    execfile(script, frozenglobals)
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/BindingGen.py", line 54, in <module>
    main()
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/BindingGen.py", line 51, in main
    generate_binding_rs(config, outputPrefix, webIDLFile)
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/BindingGen.py", line 21, in generate_binding_rs
    module = CGBindingRoot(config, outputprefix, webidlfile).define()
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/CodegenRust.py", line 6521, in __init__
    for d in dictionaries])
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/CodegenRust.py", line 6182, in __init__
    for member in dictionary.members]
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/CodegenRust.py", line 755, in getJSToNativeConversionInfo
    default = handleDefaultNull("None")
  File "/Users/ferjm/dev/mozilla/servo/components/script/dom/bindings/codegen/CodegenRust.py", line 677, in handleDefaultNull
    raise TypeError("Can't handle non-null default value here")
TypeError: Can't handle non-null default value here
```

**Comment by KiChjang** [comment]
This may have been solved by #22084. Should add a test case in TestBinding.webidl to ensure this works correctly.

**Comment by KiChjang** [comment]
Ah, so the default value for that particular field is `"interactive"`. Here's the WebIDL in question:

```webidl
dictionary AudioContextOptions {
  (AudioContextLatencyCategory or double) latencyHint = "interactive";
  float sampleRate;
};

enum AudioContextLatencyCategory {
    "balanced",
    "interactive",
    "playback"
};
```

And the codegen pukes at `(AudioContextLatencyCategory or double) latencyHint = "interactive";`, because `"interactive"` is not a default value that is null, nor is it an empty sequence (i.e. `[]`).

The complexity here is that we need to ensure that the default value falls into the set of possible values as described by the field's type during codegen, e.g. in this case, we must ensure that `"interactive"` is a valid value to the type of `(AudioContextLatencyCategory or double)` whilst we're generating the rust bindings, and if it does not, we fail with an appropriate error message during codegen.
